### ISSUE TITLE
question about plotting a meshgrid

### ISSUE THREAD

**Issue by geraldstanje** [vispy/vispy#1206]
Hi,

I currently plot my data (d.shape() = (1512, 272)) using numpy and matplotlib. How can I translate that to vispy?

```
  # 3d surface plot
  fig = plt.figure()
  ax = fig.add_subplot(111, projection='3d')

  x = np.arange(0,d.shape[1])
  y = np.arange(0,d.shape[0])
  X, Y = np.meshgrid(x, y)

  ax.plot_surface(X, Y, d, cmap=plt.cm.jet, rstride=1, cstride=1, linewidth=0)

  ax.set_xlabel('Scaling Coefficients of all features')
  ax.set_ylabel('window number')
  ax.set_zlabel('Scaling Coefficient value')

  plt.show()
```

**Comment by scottpaine** [comment]
It's possible, but not simple.

For a basic way to see only the shape plot, with no axes/colormaps/axes labels, you can do the following:

```python
import vispy.plot as vplot
x = np.arange(0,d.shape[1])
y = np.arange(0,d.shape[0])

fig = vplot.Fig()
plt = fig[0, 0]
# had to get a little backwards here due to d's shape
plt.surface(x=y, y=x, zdata=d)

if __name__ == '__main__':
    fig.show(run=True)
```

However, drawing a 3D surface with axes all labeled is not yet available in VisPy as far as I can tell. The closest you can get is by altering the code in [this example](https://github.com/vispy/vispy/blob/master/examples/basics/scene/surface_plot.py). Uncommenting lines 34-36 gives you the option to set a colormap where `cf` is set, and the closest to jet in VisPy is 'coolwarm', I think. This example does NOT include any axis labels or a z-axis at all. You'd have to set up your own axes and their corresponding labels yourself.

If you are okay with viewing your 3D object as a 2D image, the following may make your life easy:

```python
import vispy.plot as vplot

fig = vplot.Fig()
plt = fig[0, 0]
img = plt.image(d, cmap='coolwarm')
plt.xlabel.text = 'Scaling Coefficients of all features'
plt.ylabel.text = 'window number'
# may want to manually set clim here - if it's near zero, 
# you may end up with something ugly like 6.123894358976e-12 instead
plt.colorbar(position='left',
             cmap=img.cmap,
             label='Scaling Coefficient value',
             clim=(d.min(), d.max()))

if __name__ == '__main__':
    fig.show(run=True)
```

This method is also the fastest method for generating a figure.

Basically, if you want all the matplotlib stuff all together, you're gonna need to do the legwork to implement it. If you're alright with sacrificing a dimension to view the data as 2D, you'll be good to use `plt.image`. If you need the 3rd dimension but are okay with having no colormap/textlabels/axes, the first thing I wrote in here should work.

**Comment by kmuehlbauer** [comment]
@geraldstanje Did this answer your question? Please reopen, if needed.

**Comment by geraldstanje** [comment]
thanks, i didnt had time to try it out ;)
